src/Symfony/Component/Form/Event/PreSubmitEvent.php

```diff
@@ -19,6 +19,8 @@
  * It can be used to:
  *  - Change data from the request, before submitting the data to the form.
  *  - Add or remove form fields, before submitting the data to the form.
+ *
+ * @extends FormEvent<array<string, mixed>>
```

Comment:
Doesn’t this mean `PreSubmitEvent::getForm` returns `FormInterface<array<string, mixed>>`?

Reply:
Yes, as this form contains request data, which hasn't been submitted yet, so data class wasn't constructed at this point.